File: backend/matcher-service/src/main/java/com/tutor/matcher/matcher/controller/MatcherController.java
package com.tutor.matcher.matcher.controller;

import java.util.List;

import com.tutor.matcher.matcher.dto.CriteriaDto;
import com.tutor.matcher.matcher.dto.TeacherAvailabilityDto;
import com.tutor.matcher.matcher.dto.UserDto;
import com.tutor.matcher.matcherLogic.service.MatcherService;

import io.micronaut.http.annotation.Controller;
import io.micronaut.http.annotation.Get;
import io.micronaut.http.annotation.Post;
import io.swagger.v3.oas.annotations.tags.Tag;

@Tag(name = "Matcher")
@Controller("/matcher")
public class MatcherController {
	
	@Post
    List<UserDto> lookForMatch(CriteriaDto criteriaDto) {
        return MatcherService.lookForMatch(criteriaDto);
    }
	
	@Post("/info")
	List<TeacherAvailabilityDto> lookForCompleteMatch(CriteriaDto criteriaDto){
		return MatcherService.lookForCompleteMatch(criteriaDto);
	}
}

File: backend/matcher-service/src/main/kotlin/com/tutor/matcher/matcher/controller/MatcherController.kt
package com.tutor.matcher.matcher.controller

import com.tutor.matcher.dto.CriteriaDto
import com.tutor.matcher.dto.LessonDto
import com.tutor.matcher.dto.UserDto
import io.micronaut.http.HttpResponse
import io.micronaut.http.annotation.Controller
import io.micronaut.http.annotation.Get
import io.swagger.v3.oas.annotations.tags.Tag

const val MATCHER_URI = "/matcher"

@Tag(name = "Matcher")
@Controller(MATCHER_URI)
class MatcherController {
    @Get
    fun lookForMatch(criteriaDto: CriteriaDto): List<UserDto>? {
        return null;
    }
}